### PR TITLE
DEPR: remove some more Int/UInt/Float64Index from tests

### PR DESCRIPTION
**Pull request by topper-123**
Progress towards #42717, one easy step at a time... 

There are still some places that have these indexes, but maybe easier taking these things in steps.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
pandas/tests/indexes/test_base.py::TestIndex::test_reindex_doesnt_preserve_type_if_target_is_empty_index_numeric

**Comment by topper-123** [comment]
I’ve updated the PR.

File: pandas/tests/io/pytables/test_put.py [hunk 1]
```diff
@@ -280,9 +276,7 @@ def test_column_multiindex(setup_path):
 
     # non_index_axes name
     df = DataFrame(np.arange(12).reshape(3, 4), columns=Index(list("ABCD"), name="foo"))
-    expected = df.copy()
-    if isinstance(expected.index, RangeIndex):
-        expected.index = Int64Index(expected.index)
+    expected = df.set_axis(df.index.to_numpy())
```
[the remark this answers is not in the document]
Comment (topper-123) [reply pandas/tests/io/pytables/test_put.py:279]: df will always have a RangeIndex, so not needed to check. The new line is just one way to get a index with int64 dtype, there are also other ways to do it.

File: pandas/tests/indexes/test_base.py [hunk 2]
```diff
@@ -1148,10 +1139,19 @@ def test_reindex_doesnt_preserve_type_if_target_is_empty_index(self, labels, dty
         index = Index(list("abc"))
         assert index.reindex(labels)[0].dtype.type == dtype
 
+    def test_reindex_doesnt_preserve_type_if_target_is_empty_index_numeric(
```
Comment (jbrockmendel) [inline pandas/tests/indexes/test_base.py:1142]: is this moved from somewhere else or a new test?

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Pull request merged**